src/shared/directives/conditional-show/conditional-show-form.directive.ts

```diff
@@ -0,0 +1,158 @@
+      if (cf.show_condition) {
+        const condResult = ShowCondition.getInstance(cf.show_condition).match(this.allFieldValues, this.buildPath(component, cf));
+        if (cf.hidden === null || cf.hidden === undefined) {
+          cf.hidden = false;
+        }
+        if (condResult === cf.hidden) {
+          if (cf.hidden) {
+            this.greyBarService.addToggledToShow(cf.id);
+          } else {
+            this.greyBarService.removeToggledToShow(cf.id)
+          }
+          cf.hidden = !condResult;
+        }
+        // Disable the control if it's hidden so that it doesn't count towards the
+        // validation state of the form, but only if it's actually being validated.
+        if (control.validator) {
+          if (cf.hidden === true && !control.disabled) {
+            control.disable();
+          } else if (cf.hidden !== true && control.disabled) {
+            control.enable();
+          }
+        }
+      }
+    }
+  }
+
+  // make sure for the 3 callbacks that we are bound to this via an arrow function
+  private handleFormControl = (c: FormControl) => {
+    this.evaluateControl(c);
+  }
+
+  private handleFormArray = (c: FormArray, caseField: CaseField) => {
+    this.evaluateControl(c);
+    c.controls.forEach((formControl, ix) => {
+      this.fieldsUtils.controlIterator(formControl, this.handleFormArray, this.handleFormGroup, this.handleFormControl)
+   });
+  }
+
+  private handleFormGroup = (g: FormGroup) => {
+    this.evaluateControl(g);
+    let groupControl = null;
+    if (g.get('value')) { // Complex Field
+      groupControl = g.get('value') as FormGroup;
+    } else if (g.controls) {
+      // Special Fields like AddressUK, AddressGlobal
+      groupControl = g;
+    }
+    Object.keys(groupControl.controls).forEach(cKey => {
+      this.fieldsUtils.controlIterator(groupControl.get(cKey), this.handleFormArray, this.handleFormGroup, this.handleFormControl)
+    });
+  }
+
+  private evalAllShowHideConditions() {
+    this.getCurrentPagesReadOnlyAndFormFieldValues()
+    this.fieldsUtils.controlIterator(this.formGroup, this.handleFormArray, this.handleFormGroup, this.handleFormControl)
+  }
+
+  private buildPath(c: AbstractFormFieldComponent, field: CaseField) {
+    if (c && c instanceof AbstractFieldWriteComponent) {
+      if (c.idPrefix) {
+        return c.idPrefix + field.id;
+      }
+    }
+    return field.id;
+  }
+
+  private getCurrentPagesReadOnlyAndFormFieldValues() {
+    let formFields = this.getFormFieldsValuesIncludingDisabled();
+    this.allFieldValues = this.fieldsUtils.mergeCaseFieldsAndFormFields(this.contextFields, formFields);
+    return this.allFieldValues;
+  }
+
+  private getFormFieldsValuesIncludingDisabled() {
+    return this.formGroup.getRawValue();
+  }
+
+  private unsubscribeFromFormChanges() {
+    if (this.formChangesSubscription) {
+      this.formChangesSubscription.unsubscribe();
+    }
+  }
+
+  // TODO - remove or make work for a specific field
```

Comment:
Is this working or will it be removed?

Reply:
It's currently not being called by anything and I think it's just a holdover from `ConditionalShowDirective`. I don't think it's needed here. @andywilkinshmcts to confirm whether the method can be deleted.

Reply:
Thanks, @paul-graham - will await @andywilkinshmcts's confirmation.